### PR TITLE
Add Ordos 6a

### PR DESCRIPTION
**Pull request by MustaphaTR**
I had this almost done for a long while now. I got an idea to solve the small problem that made me leave it so handled that, done other stuff remaining and here is the mission.

May do 6b soon too.

### CODE REVIEW

**Comment by Smittytron** [comment]
"The Atreides have been annihilated!" message repeats eternally.

**Comment by MustaphaTR** [comment]
Updated.

**Comment by Smittytron** [comment]
Still getting an infinite loop on the message.

Also, I'm skeptical that hard difficulty is beatable as it is.

**Comment by abcdefg30** [comment]
What is the status here?

File: mods/d2k/maps/ordos-06a/ordos06a.lua [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+Tick = function()
+	playerStarports = player.GetActorsByType("starport")
+
+	if #playerStarports < 1 and not player.IsObjectiveFailed(GuardStarport) then
+		player.MarkFailedObjective(GuardStarport)
+		atreides.MarkCompletedObjective(KillStarport1)
+		harkonnen.MarkCompletedObjective(KillStarport2)
+	end
+
+	if atreides.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillHarkonnen) then
```
Comment (ltem) [inline mods/d2k/maps/ordos-06a/ordos06a.lua]: In this line `KillHarkonnen` should be changed to `KillAtreides` else it is not possible to finish the mission if the player defeats the Harkonnen first.
```diff
-       if atreides.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillHarkonnen) then
+       if atreides.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillAtreides) then
                Media.DisplayMessage("The Atreides have been annihilated!", "Mentat")
                player.MarkCompletedObjective(KillAtreides)
        end
```

**Comment by ltem** [comment]
I still need to play the replay the original mission but except the OpenRA-Starport-Behavior I didn't notice any differences so far.

I was not able to beat the mission on `Hard`, which is very difficult but it can also be that I just don't have enough practice.

File: mods/d2k/maps/ordos-06a/ordos06a.lua (hunk 1 above)
Comment (reaperrr) [reply mods/d2k/maps/ordos-06a/ordos06a.lua]: That should also fix the looping message.

**Comment by ltem** [comment]
I finally played through the original missions on `Normal` and `Hard` and noticed that the attack frequency of the enemies is much lower than in the OpenRA version. You can cope with it on `Normal` but on `Hard` it is in my opinion too much (in other words I wasn't able to beat the `Hard` OpenRA version).  Maybe the carryall reinforcements have a higher delay between the attacks?!

Also the duration till the Ordos reinforcements arrive is always 30 minutes for every difficulty level in the original mission. Is there a reason why you chose the 20/30/40 minutes delay? 40 minutes is a quite long duration.

Additional, there is an `Enemey reinforcements have arrived.`-message (including an audio signal) when the carryall flies in to drop enemy units in the original mission, which could help to warn player that an attack wave is incoming. I don't know if you want to add this or how much effort it would be, I just want to mention that.
However, there seems to be no warning for the regular attacks of Harkonnen and Atreides, which trickle in every now and then.

|image|
|:---:|
|Attack warning in original mission.|

So fixing the `yaml` error above and reducing the difficulty for `Hard` (+Rebasing) would fix the PR for me.

PS: An other benefit of the original mission is that the fog of war won't reappear, so the map can be scouted and allows to spot the enemies attack direction earlier but in my understanding this is not something, which will be added in OpenRA and I'm fine with that.

**Comment by MustaphaTR** [comment]
How OpenRA D2k mission reinforcements work is pretty different than original, not just for this mission. Maybe a compalte overhaul to change them to be as in original could be better.

I think it is better that difficulty affects the timer. Tho if you think 40 mins is too much, i can change it to 25,30,35.

**Comment by ltem** [comment]
> How OpenRA D2k mission reinforcements work is pretty different than original, not just for this mission. Maybe a compalte overhaul to change them to be as in original could be better.

That sounds like a large PR for the future. I think for this mission it is enough to increase the delay between the enemy attacks at the beginning of the mission. Or am I misunderstanding you?

I also noticed that if the countdown reaches 1:20 an extra strong enemy reinforcement army arrives and attacks the player.  Do the original mission files contain any information about this?

> I think it is better that difficulty affects the timer. Tho if you think 40 mins is too much, i can change it to 25,30,35.

Yep, that sounds better. In my case I noticed that after certain time my army is strong enough to withstand the enemies in whatever configuration they come. From this point on I'm only waiting till the timer finishes to get the reinforcements. So very large countdowns are kind of boring. 
Actually the D2K infantry in OpenRA is effective enough to annihilate the enemy bases on their own^^ but I didn't find out why they are that capable.

**Comment by abcdefg30** [comment]
> Maybe a compalte overhaul to change them to be as in original could be better.

I disagree (at least for the Atreides missions). I got times from the original files (using penevs 'editor') and adjusted them as I saw fit. The aim here is not to have an exact 1:1 port from the original, but a fun mission to play.

**Comment by reaperrr** [comment]
It's a bit of a shame this didn't make it into the upcoming release, let's try to at least get this into Next+1.

**Comment by reaperrr** [comment]
Any updates on the horizon @MustaphaTR?

**Comment by abcdefg30** [comment]
I can take this over in case you're no longer interested or don't have the time to update the PR.

**Comment by MustaphaTR** [comment]
@abcdefg30 Please do, i don't have much time anymore.

**Comment by abcdefg30** [comment]
Superseded by #17580.

**Pull request closed without merging**